From 8339c2ba0f8cf509c5a6fead1f45078529554a5c Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Mon, 5 Dec 2022 15:12:47 +0800
Subject: [PATCH] s32: hse: support secure boot feature on both S32G2 and S32G3
 platform

When installing the core reset entry into HSE module, the cortex A53_0 core
ID of S32G2 SoC is different from the one in S32G3 SoC. The one in  S32G2 is
3 and the one in S32G3 is 4. So improve code to support secure boot feature
on both S32G2 and S32G3 platform

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-s32/s32-cc/hse_adv_secboot.c | 6 ++++++
 include/hse/hse_abi.h                      | 3 ++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c b/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c
index 18d8768ac3..cf5b21b99e 100644
--- a/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c
+++ b/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c
@@ -227,7 +227,13 @@ int hse_install_cr_entry(struct hse_private *priv, u32 *recv_buf)
 	memset((void *)&priv->cr_entry, 0, sizeof(struct hse_cr_entry));
 	memset((void *)&priv->srv_desc, 0, sizeof(struct hse_srv_desc));
 
+#if defined(CONFIG_ARCH_S32G2)
 	priv->cr_entry.core_id = HSE_APP_CORE3;
+#elif defined(CONFIG_ARCH_S32G3)
+	priv->cr_entry.core_id = HSE_APP_CORE4;
+#else
+	priv->cr_entry.core_id = HSE_APP_CORE3;
+#endif
 	priv->cr_entry.cr_sanction = HSE_CR_SANCTION_KEEP_CORE_IN_RESET;
 	priv->cr_entry.preboot_smr_map = HSE_SMR_ENTRY_1;
 	priv->cr_entry.pass_reset = priv->app_boot_hdr.ram_entry;
diff --git a/include/hse/hse_abi.h b/include/hse/hse_abi.h
index 1233e4b859..de5ed212de 100644
--- a/include/hse/hse_abi.h
+++ b/include/hse/hse_abi.h
@@ -51,7 +51,8 @@
 
 #define HSE_ACCESS_MODE_ONE_PASS 0u
 
-#define HSE_APP_CORE3 3u /* A53_0 */
+#define HSE_APP_CORE3 3u /* A53_0 of S32G2 SoC */
+#define HSE_APP_CORE4 4u /* A53_0 of S32G3 SoC */
 
 #define HSE_PUBLISH_UPDATED_DATA_SET BIT(0)
 #define HSE_PUBLISH_ALL_DATA_SETS    BIT(1)
-- 
2.36.1

